modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts

```diff
@@ -16,94 +16,167 @@ import {
-	const valuesArray = arrayOfObjects?.map((values: string) => {
+	const valuesArray = arrayOfObjects?.map((values: any) => {
 		return Object.values(values)[0];
 	});
 
 	return valuesArray;
 }
 
-function populateGoals(goalsResult: string[], goalsArray: string[]) {
-	goalsResult.forEach((policy: any) => {
-		const month = new Date(policy?.finalReferenceDate)
+function populateGoalsProducts(
+	goalsResult: SalesGoal[],
+	goalsArray: string[],
+	productExternalReferenceCode: string
+) {
+	goalsResult.forEach((policy: SalesGoal) => {
+		const month = new Date(policy.finalReferenceDate)
 			.toUTCString()
 			.split(' ')[2];
-
 		goalsArray?.forEach((goalElement: any) => {
 			if (month in goalElement) {
-				goalElement[month] += policy?.goalValue;
+				if (
+					policy?.productExternalReferenceCode ===
+					productExternalReferenceCode
+				) {
+					goalElement[month] = policy?.goalValue;
+				}
+				else {
+					goalElement[month] += policy?.goalValue;
+				}
 			}
 		});
 	});
 
 	return goalsArray;
 }
 
-function populateSales(policiesResult: string[], policiesArray: string[]) {
-	policiesResult.forEach((policy: any) => {
+function populateSalesProducts(
+	policiesResult: SalesPolices[],
+	policiesArray: string[],
+	productExternalReferenceCode: string
+) {
+	policiesResult.forEach((policy: SalesPolices) => {
 		const month = new Date(policy?.boundDate).toUTCString().split(' ')[2];
-
 		policiesArray?.forEach((policyElement: any) => {
 			if (month in policyElement) {
-				policyElement[month] += policy?.termPremium;
+				if (
+					policy?.productExternalReferenceCode ===
+					productExternalReferenceCode
+				) {
+					policyElement[month] += policy?.termPremium;
+				}
+				else {
+					policyElement[month] += policy?.termPremium;
+				}
 			}
 		});
 	});
 
 	return policiesArray;
 }
 
-const getArrayOfSales = (response: any, arrayOfMonthsArray: string[]) => {
-	const monthsResult = response?.data?.items;
-	const arrayOfMonths = populateSales(monthsResult, arrayOfMonthsArray);
+const getArrayOfSalesProducts = (
+	response: SalesPolices[],
+	arrayOfMonthsArray: string[],
+	productExternalReferenceCode: string
+) => {
+	const monthsResult = response;
+	const arrayOfMonths = populateSalesProducts(
+		monthsResult,
+		arrayOfMonthsArray,
+		productExternalReferenceCode
+	);
 
 	return getValuesFromArrayOfObjects(arrayOfMonths);
 };
-const getArrayOfGoals = (response: any, monthsAgoGoalsArray: string[]) => {
-	const monthsGoalsResult = response?.data?.items;
-	const monthsAgoGoals = populateGoals(
+
+const getArrayOfGoalsProducts = (
+	response: SalesGoal[],
+	monthsAgoGoalsArray: string[],
+	productExternalReferenceCode: string
+) => {
+	const monthsGoalsResult = response;
+	const monthsAgoGoals = populateGoalsProducts(
 		monthsGoalsResult,
-		monthsAgoGoalsArray
+		monthsAgoGoalsArray,
+		productExternalReferenceCode
 	);
 
 	return getValuesFromArrayOfObjects(monthsAgoGoals);
 };
 
+const compareProducts = async (
```

Comment:
Why do you use async here? Is it necessary?

```suggestion
const compareProducts = (
```